src/main/java/net/mcreator/ui/modgui/ItemExtensionGUI.java

```diff
@@ -0,0 +1,197 @@
+
+		fuelPanel.setBorder(BorderFactory.createTitledBorder(
+				BorderFactory.createLineBorder((Color) UIManager.get("MCreatorLAF.BRIGHT_COLOR"), 1),
+				L10N.t("elementgui.item_extension.fuel_properties"), TitledBorder.LEADING,
+				TitledBorder.DEFAULT_POSITION, getFont(), (Color) UIManager.get("MCreatorLAF.BRIGHT_COLOR")));
+
+		// Compostable
+		JPanel compostPanel = (JPanel) PanelUtils.westAndCenterElement(
+				HelpUtils.wrapWithHelpButton(this.withEntry("item_extension/layer_chance"),
+						L10N.label("elementgui.item_extension.layer_chance")), layerChance);
+		compostPanel.setBorder(BorderFactory.createTitledBorder(
+				BorderFactory.createLineBorder((Color) UIManager.get("MCreatorLAF.BRIGHT_COLOR"), 1),
+				L10N.t("elementgui.item_extension.compost_properties"), TitledBorder.LEADING,
+				TitledBorder.DEFAULT_POSITION, getFont(), (Color) UIManager.get("MCreatorLAF.BRIGHT_COLOR")));
+
+		// Dispenser behaviour
+		dispenseSuccessCondition = new ProcedureSelector(this.withEntry("item_extension/dispense_success_condition"),
+				mcreator, L10N.t("elementgui.item_extension.dispense_success_condition"),
+				VariableTypeLoader.BuiltInTypes.LOGIC, Dependency.fromString(
+				"x:number/y:number/z:number/world:world/itemstack:itemstack/direction:direction")).makeInline();
+		dispenseResultItemstack = new ProcedureSelector(this.withEntry("item_extension/dispense_result_itemstack"),
+				mcreator, L10N.t("elementgui.item_extension.dispense_result_itemstack"),
+				VariableTypeLoader.BuiltInTypes.ITEMSTACK, Dependency.fromString(
+				"x:number/y:number/z:number/world:world/itemstack:itemstack/direction:direction/success:boolean")).setDefaultName(
+						L10N.t("elementgui.item_extension.dispense_result_itemstack.default")).makeInline()
+				.makeReturnValueOptional();
+
+		JComponent canDispense = PanelUtils.gridElements(1, 2, 0, 5,
+				HelpUtils.wrapWithHelpButton(this.withEntry("item_extension/has_dispense_behavior"),
+						L10N.label("elementgui.item_extension.has_dispense_behavior")), hasDispenseBehavior);
+		JComponent dispenseProcedures = PanelUtils.gridElements(2, 1, 0, 2, dispenseSuccessCondition,
+				dispenseResultItemstack);
+
+		hasDispenseBehavior.setOpaque(false);
+		hasDispenseBehavior.setSelected(false);
+		hasDispenseBehavior.addActionListener(e -> updateDispenseElements());
+
+		JComponent dispenserBehaviourPanel = PanelUtils.northAndCenterElement(canDispense,
+				PanelUtils.centerInPanel(dispenseProcedures));
+
+		dispenserBehaviourPanel.setBorder(BorderFactory.createTitledBorder(
+				BorderFactory.createLineBorder((Color) UIManager.get("MCreatorLAF.BRIGHT_COLOR"), 1),
+				L10N.t("elementgui.item_extension.dispense_properties"), TitledBorder.LEADING,
+				TitledBorder.DEFAULT_POSITION, getFont(), (Color) UIManager.get("MCreatorLAF.BRIGHT_COLOR")));
+		dispenserBehaviourPanel.setOpaque(false);
+
+		addPage(PanelUtils.totalCenterInPanel(PanelUtils.northAndCenterElement(PanelUtils.join(

```

Comment:
This would be more readable if you split all this statement into variables with smaller panels.